genai/files.go

```diff
@@ -83,6 +84,19 @@ func (c *Client) UploadFile(ctx context.Context, name string, r io.Reader, opts
 	return c.GetFile(ctx, res.File.Name)
 }
 
+// UploadFileFromPath is a convenience method wrapping [UploadFile]. It takes
+// a path to read the file from, and uses a default auto-generated ID for the
+// uploaded file.
+func (c *Client) UploadFileFromPath(ctx context.Context, path string, opts *UploadFileOptions) (*File, error) {
```

Comment:
Isn't there a loop waiting for the file to be ready?

Reply:
Yes, but according to the API folks it's not really necessary for non-video. A loop can still be added around `UploadFileFromPath` if needed, and they asked us to do it explicitly in the relevant sample (https://github.com/google/generative-ai-go/blob/main/genai/internal/samples/docs-snippets_test.go#L1202)

This aligns with the Python SDK and samples, where `upload_file` doesn't do the looping itself, but you can do it on your own for videos: https://github.com/google-gemini/generative-ai-python/blob/main/samples/files.py#L68